### PR TITLE
Sdl client

### PR DESCRIPTION
**Pull request by akallabeth**
This pr adds a simple `SDL2` based client.
The idea is to unify all our platform dependent implementations so we only have a single client to maintain for all platforms (or at least have one)
that way we would have something feature equivalent on all supported environments and not the state we currently have (every platform has very different features/bugs)

What already works: 
* Single window session
* mouse input
* keyboard input
* touch input
* dynamic resolution (window mode)
* AuthOnly support
* fullscreen
 
What is still to be done:
* dynamic display add/remove/resize
* keyboard capture
* hotkeys (capture input, toggle fullscreen, ...)
* multimon
* RAILS

### CODE REVIEW

File: client/Wayland/wlf_input.c [hunk 1]
```diff
@@ -396,197 +396,57 @@ BOOL wlf_keyboard_modifiers(freerdp* instance, const UwacKeyboardModifiersEvent*
...
-		WINPR_ASSERT(rdpei->TouchBegin);
-		rdpei->TouchBegin(rdpei, touchId, x, y, &contactId);
-	}
-#else
-	WLog_WARN(TAG, "Touch event detected but RDPEI support not compiled in. Recompile with "
-	               "-DWITH_CHANNELS=ON");
-#endif
-
-	return TRUE;
+    return freerdp_client_handle_touch(&wlf->common, FREERDP_TOUCH_DOWN, ev->id, 0,x, y);
```
Comment (hardening) [inline client/Wayland/wlf_input.c:432]: missing space after `0,`

File: client/common/client.c [hunk 2]
```diff
@@ -1358,6 +1360,192 @@ BOOL freerdp_client_send_extended_button_event(rdpClientContext* cctx, BOOL rela
...
+
+	// Emulate mouse click if touch is not possible, like in login screen
+	if (!rdpei)
+	{
+		UINT16 flags = 0;
+		flags |= PTR_FLAGS_DOWN;
+		flags |= PTR_FLAGS_MOVE;
+		flags |= PTR_FLAGS_BUTTON1;
+
+        WINPR_ASSERT(contact->x <= UINT16_MAX);
```
Comment (hardening) [inline client/common/client.c:1424]: strange code indentation here

File: client/common/client.c [hunk 3]
```diff
@@ -1358,6 +1360,192 @@ BOOL freerdp_client_send_extended_button_event(rdpClientContext* cctx, BOOL rela
...
+	WINPR_ASSERT(cctx);
+	WINPR_ASSERT(ppcontact);
+
+	for (size_t i = 0; i < ARRAYSIZE(cctx->contacts); i++)
+	{
+		FreeRDP_TouchContact* contact = &cctx->contacts[i];
+
+		if (contact->id == touchId)
+		{
+            *ppcontact = contact;
```
Comment (hardening) [inline client/common/client.c:1512]: indentation pb here too

File: include/freerdp/client.h [hunk 4]
```diff
@@ -79,6 +79,17 @@ extern "C"
 	typedef int (*pRdpClientEntry)(RDP_CLIENT_ENTRY_POINTS* pEntryPoints);
 
 	/* Common Client Interface */
+#define FREERDP_MAX_TOUCH_CONTACTS 10
+
+	typedef struct
+	{
+		ALIGN64 INT32 id;
+		ALIGN64 UINT32 count;
+        ALIGN64 INT32 x;
```
Comment (hardening) [inline include/freerdp/client.h:88]: indent pb here

**Review by hardening — APPROVED** [review]
Nice job

**Comment by pnowack** [comment]
Will this already have CSD (client side decorations) on Wayland or does it still require `libdecor` for this?

**Comment by akallabeth** [comment]
@pnowack well, whatever `SDL` provides

**Comment by bmiklautz** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test pretty please

**Pull request merged**